### ISSUE TITLE
Device status online when coap gateway is down

### ISSUE THREAD

**Issue by itestoss** [plgd-dev/hub#627]
### Description
When coap gateway is not running, plgd hub continues to report the previous online/offline device status of the device even though it cannot communicate with it (and so cannot be sure that is the actual state of the device).

### How can it be reproduced?

1. deploy using command:
microk8s.helm3 install am-1 -f withMock.yaml plgd/plgd-hub --create-namespace --namespace=am-1 --version 2.2.2

withMock.yaml content:
 
```
global:
  domain: "host.domain.com"
  hubId: "d03a1bb4-0a77-428c-b78c-1c46efe6a38e"
mockoauthserver:
  enabled: true
coapgateway:
  port: 5683
  service:
    type: NodePort
    nodePort: 5683
  apis:
    coap:
      tls:
        enabled: false
```

2. onboard devices and check their status

- export ACCESS_TOKEN=$(curl -ks "https://${HOSTNAME}:443/oauth/token?client_id=test&audience=test" | jq -r .access_token)

- curl -ks -XGET "https://${HOSTNAME}:443/api/v1/devices" --header 'Content-Type: application/json' --header "Authorization: Bearer ${ACCESS_TOKEN}"

3. stop coap gateway (by scaling it down to zero replicas)
microk8s.kubectl scale deployment am-1-plgd-hub-coap-gateway --replicas=0 --namespace am-1 

4. Check devices status again
- curl -ks -XGET "https://${HOSTNAME}:443/api/v1/devices" --header 'Content-Type: application/json' --header "Authorization: Bearer ${ACCESS_TOKEN}"

### Environment
- microk8s v1.22.4 (snap) running on CentOS Linux release 8.5.2111
- microk8s helm3 v3.5.0
- plgd hub tag: 2.2.2

**Comment by ondrejtomcik** [comment]
Dear @itestoss

If the pod is terminated gracefully, the CoAP Gateway shall set the state of each connected device to offline. Scaling deployment to 0 replicas is a graceful shutdown, therefore it should behave as you expect. We will look into this very soon. Stay tuned.

**Comment by jkralik** [comment]
Dear @itestoss
The fix will be released in the next version this week.

**Comment by itestoss** [comment]
Thank you 👍

**Comment by jkralik** [comment]
Dear @itestoss 
We released a new version [2.2.3](https://github.com/plgd-dev/hub/releases/tag/v2.2.3) with the fix.
